### PR TITLE
feat: GatePool CRUD

### PR DESCRIPTION
**Pull request by TheRealSeber**
Get, Put, Delete endpoints for GatePool,

if there is any another endpoint i should do let me know. Additionaly I am getting the following error on get endpoint:

```
backend-api-1       | pydantic.error_wrappers.ValidationError: 6 validation errors for Pool
backend-api-1       | response -> 0 -> name
backend-api-1       |   field required (type=value_error.missing)
backend-api-1       | response -> 0 -> description
backend-api-1       |   field required (type=value_error.missing)
backend-api-1       | response -> 0 -> created_by
backend-api-1       |   field required (type=value_error.missing)
backend-api-1       | response -> 0 -> created_at
backend-api-1       |   field required (type=value_error.missing)
backend-api-1       | response -> 0 -> updated_by
backend-api-1       |   field required (type=value_error.missing)
backend-api-1       | response -> 0 -> updated_at
backend-api-1       |   field required (type=value_error.missing)
```

not sure what is the issue, since we should return from our service GatePool object not Pool. Maybe you will find it, before me so I post it here. 

Also need a review.

### CODE REVIEW

**Comment by TheRealSeber** [comment]
image

Okay, so basically, when calling gate/{gate_id}/pool/ endpoint there are two things that are happening:
1. we are getting 404 response which indicates that there isn't any gate of a given id
2. we are **not even entering** our router if the gate exists?

I have tried to print "dupa" as you can see above but, well - it wasn't printed. Any idea what the fuck is happening? I attach full error below:

```
2023-06-30 22:12:47 [INFO] 2023-06-30 20:12:47,552 - uvicorn.access - 192.168.32.1:34940 - "GET /gate/1/pool/?skip=0&limit=100 HTTP/1.1" 500
2023-06-30 22:12:47 [ERROR] 2023-06-30 20:12:47,553 - uvicorn.error - Exception in ASGI application
2023-06-30 22:12:47 Traceback (most recent call last):
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 429, in run_asgi
2023-06-30 22:12:47     result = await app(  # type: ignore[func-returns-value]
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
2023-06-30 22:12:47     return await self.app(scope, receive, send)
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/fastapi/applications.py", line 276, in __call__
2023-06-30 22:12:47     await super().__call__(scope, receive, send)
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
2023-06-30 22:12:47     await self.middleware_stack(scope, receive, send)
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
2023-06-30 22:12:47     raise exc
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
2023-06-30 22:12:47     await self.app(scope, receive, _send)
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/starlette/middleware/sessions.py", line 86, in __call__
2023-06-30 22:12:47     await self.app(scope, receive, send_wrapper)
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
2023-06-30 22:12:47     raise exc
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
2023-06-30 22:12:47     await self.app(scope, receive, sender)
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
2023-06-30 22:12:47     raise e
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
2023-06-30 22:12:47     await self.app(scope, receive, send)
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
2023-06-30 22:12:47     await route.handle(scope, receive, send)
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
2023-06-30 22:12:47     await self.app(scope, receive, send)
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
2023-06-30 22:12:47     response = await func(request)
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/fastapi/routing.py", line 255, in app
2023-06-30 22:12:47     content = await serialize_response(
2023-06-30 22:12:47   File "/opt/conda/envs/virtualenv/lib/python3.10/site-packages/fastapi/routing.py", line 141, in serialize_response
2023-06-30 22:12:47     raise ValidationError(errors, field.type_)
2023-06-30 22:12:47 pydantic.error_wrappers.ValidationError: 6 validation errors for Pool
2023-06-30 22:12:47 response -> 0 -> name
2023-06-30 22:12:47   field required (type=value_error.missing)
2023-06-30 22:12:47 response -> 0 -> description
2023-06-30 22:12:47   field required (type=value_error.missing)
2023-06-30 22:12:47 response -> 0 -> created_by
2023-06-30 22:12:47   field required (type=value_error.missing)
2023-06-30 22:12:47 response -> 0 -> created_at
2023-06-30 22:12:47   field required (type=value_error.missing)
2023-06-30 22:12:47 response -> 0 -> updated_by
2023-06-30 22:12:47   field required (type=value_error.missing)
2023-06-30 22:12:47 response -> 0 -> updated_at
2023-06-30 22:12:47   field required (type=value_error.missing)
```

**Review by HermanPlay — CHANGES_REQUESTED** [review]
Answer my comments please.

File: app/api/services/gate_pool.py [hunk 1]
```diff
@@ -11,7 +10,43 @@ class GatePoolService:
...
+    ) -> gate_pool_models.GatePool:
+        """
+        Returns the pool in the given gate pool data found by pool ID
+
+        :param gate_id: the given gate pool ID to where to look for the pool ID
+        :param pool_id: the pool ID to retrieve
+
+        :return: the gate pool data corresponding to the given IDs or None if not found
+        """
+        return (
```
[the remark this answers is not in the document]
Comment (TheRealSeber) [reply app/api/services/gate_pool.py:24]: yep, it works as expected

**Comment by TheRealSeber** [comment]
nevermind, i deleted local docker images, and made a rebuild => it started to work kek

**Review by matined — APPROVED** [review]

**Comment by HermanPlay** [comment]
Very good job, Mr S

**Review by HermanPlay — APPROVED** [review]

**Pull request merged**